### ISSUE TITLE
Support for the Gradebook MFE

### ISSUE THREAD

**Issue by arbrandes** [openedx/build-test-release-wg#29]
Coordinate, and if necessary execute, the tasks necessary to get the [Gradebook MFE](https://github.com/edx/frontend-app-gradebook) up to speed for Lilac:

1. It passes a minimum standard of documentation: at the very least, the READMEs should have descriptive text that includes use cases, screenshots, and particularly, documentation on that MFE's environment variables.
2. Its basic feature-set (as documented above) works reliably with the rest of the Lilac codebase.
3. It's deployed by default using the Native Installation, and deployable as a plugin using Tutor.
4. It is internationalized (i18n), localizable (l10n), and passes minimum accessibility (a11y) standards.
5. It has a header and footer that are customizable via npm aliases: https://edx.readthedocs.io/projects/edx-developer-docs/en/latest/developers_guide/micro_frontends_in_open_edx.html#overriding-brand-specific-elements
6. It is compatible with the @edx/brand system: https://open-edx-proposals.readthedocs.io/en/latest/oep-0048-brand-customization.html
7. It uses CSS utility classes in a branding-friendly way. (i.e., don't use things like 'bg-white')

**Comment by arbrandes** [comment]
@regisb, can we count on you for Tutor support for this, as well as for #28 and #27?

**Comment by regisb** [comment]
Yes, the required MFEs will be delivered for Tutor in v12.

**Comment by sarina** [comment]
@arbrandes -- given the MFE is currently not internationalized, are we going to put this MFE aside for the more critical ecommerce MFEs?

cc @kdmccormick

**Comment by arbrandes** [comment]
@sarina, apologies for the protracted delay, but as we've already discussed in the meeting, yup, we're dropping this one.

**Comment by jnlapierre** [comment]
Good Morning :) I am going to be reviewing this.  I plan to review it today.

**Comment by nedbat** [comment]
@jnlapierre is the edX contact for this work. Thanks Justin!

**Comment by jfavellar90** [comment]
Thanks @nedbat, @jnlapierre Please let me know if you require any help in the review you're going to perform.

**Comment by jfavellar90** [comment]
Hi @jnlapierre did you have the chance to review the Gradebook MFE?

**Comment by jfavellar90** [comment]
@davidjoy @jnlapierre hi, do you have any updates related to the Gradebook MFE?

**Comment by jfavellar90** [comment]
@davidjoy @jnlapierre  I added a PR to fix the problem of the courseId calculation [here](https://github.com/edx/frontend-app-gradebook/pull/213). Could you please help me with a review? Thanks!

**Comment by jfavellar90** [comment]
@BbrSofiane I think this is not ready to be marked as DONE. I'll be working on checking the branding, header and footer overrides and i18n

**Comment by BbrSofiane** [comment]
@jfavellar90 yes, I meant to close the other issue.

**Comment by jfavellar90** [comment]
Comments related to this task:

- I can confirm it is compatible with the @edx/brand system. I used the patch `mfe-dockerfile-post-npm-install` offered by [Tutor](https://github.com/overhangio/tutor-mfe/blob/a8bb82c272f778315ea8f940cad358144e175875/tutormfe/templates/mfe/build/mfe/Dockerfile#L40) to install a fork of the edx/brand-openedx repo (https://github.com/eduNEXT/brand-openedx/) with a couple of style changes. It worked as expected. The line to install the custom branding repo was the following]:

`RUN npm install --no-audit --no-fund @edx/brand@git+https://github.com/eduNEXT/brand-openedx.git --save`

- I had issues running the footer override. I just forked the [base repo](https://github.com/edx/frontend-component-footer) and applied a couple of changes on the last tagged version (v10.1.6). Then tried to install the footer with npm aliases through the same `mfe-dockerfile-post-npm-install` patch. When building the grade book app with the override I get the error [HERE](https://gist.github.com/jfavellar90/12b10bd5462382cfc356edc32f41d7a3)
Doing a little research I found [this post](https://discuss.openedx.org/t/mfe-header-override/5125) from where I took a couple of ideas to fix the issue. Specifically, I tried [this](https://discuss.openedx.org/t/mfe-header-override/5125/7) and [this](https://discuss.openedx.org/t/mfe-header-override/5125/5), obtaining the same failure. @davidjoy could you please take a look? I suspect this is something easy to fix.
This MFE does not support header override. We need to decide if this is a must-have for the final release.

- The [PR](https://github.com/edx/frontend-app-gradebook/pull/213) related to the course extraction fix is still waiting for a review. I already tagged again the team in charge of the repo.

- Related to i18n, @regisb has been working on creating the project in Transifex for this MFE and created a [PR](https://github.com/edx/frontend-app-gradebook/pull/217) with the first version of the translations pulled from Transifex.

**Comment by davidjoy** [comment]
I'm pretty sure the import shouldn't be prefixed with an underscore `_`:

```
@import "~@edx/frontend-component-footer/dist/_footer";
```

It's a fragment, so SASS should find it without that.

**Comment by pdpinch** [comment]
FYI, regarding the header, @asadiqbal08 has the follow PRs in-flight for the Learning MFE:

https://github.com/edx/frontend-app-learning/pull/715
https://github.com/edx/frontend-component-header/pull/133
https://github.com/edx/frontend-component-header-edx/pull/196

**Comment by jfavellar90** [comment]
@davidjoy According to my testing, the problem with the footer override is that `dist` folder is not included in the GitHub repos. I had to manually stop ignoring the `dist` folder in my footer component fork (https://github.com/eduNEXT/frontend-component-footer/commit/89015dc1fec7da89313b0989181bc7463216c7ba) and install the component from the repo. Once this is done, the footer override works as expected. I'm not pretty sure this is a good practice but it was the only way I got it working. any thoughts related to this approach?

**Comment by BbrSofiane** [comment]
> Once this is done, the footer override works as expected. I'm not pretty sure this is a good practice but it was the only way I got it working. any thoughts related to this approach?

@davidjoy any thoughts on this?

**Comment by BbrSofiane** [comment]
@jfavellar90 is this the last hurdle to for the gradebook MFE to be ready?

**Comment by jfavellar90** [comment]
@BbrSofiane we need to get merged this PR: https://github.com/edx/frontend-app-gradebook/pull/213 in order to make the MFE work properly in tutor, @davidjoy is there any way to accelerate this process? Once merged it must be backported to Gradebook maple release

I have another PR in https://github.com/edx/frontend-app-gradebook/pull/220 related to the `frontend-component-header` implementation, but this is not a must-have for the Maple release (this PR can take a bit longer to get merged).

On the other hand, we need to backport the [language work](https://github.com/edx/frontend-app-gradebook/pull/217) that Regis already got merged in master. I was wondering if we can reset maple release branch (`open-release/maple.beta1`) to the current master.

Summarizing, the minimum required for the release is:
- Merge and backport of the PR https://github.com/edx/frontend-app-gradebook/pull/213
- Backport of the Regis [language work](https://github.com/edx/frontend-app-gradebook/pull/217)

I remain super attentive to any feedback.

**Comment by BbrSofiane** [comment]
edx/frontend-app-gradebook#218 - backport to maple

**Comment by jfavellar90** [comment]
@BbrSofiane backport course id issue https://github.com/edx/frontend-app-gradebook/pull/221

**Comment by BbrSofiane** [comment]
merge edx/frontend-app-gradebook#222
